textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexToGramQueryTranslator.java

```diff
@@ -74,6 +53,12 @@ static GramBooleanQuery translateUnsimplified(String regex, int gramLength)
 
         TranslatorUtils.GRAM_LENGTH = TranslatorUtils.DEFAULT_GRAM_LENGTH;
 
+        TranslatorUtils.escapeSpecialCharacters(regexInfo.match);
+
+        // Since the inverted index relies on lower-case grams, we need to
+        // convert the characters to lower case.
+        TranslatorUtils.toLowerCase(regexInfo.match);
```

Comment:
It's literally the same to do it before translation or after translation. Doing it before translation may save us a little bit of time, since we maintain a set of grams.

@jesmaelnezhad you can change it to do `regex.tolowercase` in the beginning of the translation.